Fix #11220: Restore @seanlip ownership, remove @BenHenning temporarily, and add Oppia ML owners

BenHenning opened the pull request:
## Overview
<!--
READ ME FIRST:
Please answer *both* questions below and check off every point from the Essential Checklist!
If there is no corresponding issue number, fill in N/A where it says [fill_in_number_here] below in 1.
-->

1. This PR fixes #11220.
2. This PR does the following: Restore code ownership to @seanlip & remove @BenHenning until 2021-01-04 (per #11477). This also introduces global ownership as a fallback in case new files are added that aren't covered by other owners, and replaces Prasanna with a load-balanced review team: @oppia/ml.

## Essential Checklist

- [X] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [X] The linter/Karma presubmit checks have passed locally on your machine.
- [ ] "Allow edits from maintainers" is checked. (See [here](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) for instructions on how to enable it.)
  - This lets reviewers restart your CircleCI tests for you.
- [X] The PR is made from a branch that's **not** called "develop".

## PR Pointers

- Make sure to follow the [instructions for making a code change](https://github.com/oppia/oppia/wiki/Contributing-code-to-Oppia#instructions-for-making-a-code-change).
- Oppiabot will notify you when you don't add a PR_CHANGELOG label. If you are unable to do so, please @-mention a code owner (who will be in the Reviewers list), or ask on [Gitter](https://gitter.im/oppia/oppia-chat).
- For what code owners will expect, see the [Code Owner's wiki page](https://github.com/oppia/oppia/wiki/Oppia%27s-code-owners-and-checks-to-be-carried-out-by-developers).
- Make sure your PR follows conventions in the [style guide](https://github.com/oppia/oppia/wiki/Coding-style-guide), otherwise this will lead to review delays.
- Never force push. If you do, your PR will be closed.

BenHenning committed:
Fix #11220: Restore code ownership to @seanlip & remove @BenHenning temporarily

Fix #11220

Restore code ownership to @seanlip & remove @BenHenning until 2021-01-04 (per #11477).

BenHenning commented:
@rt4914 PTAL as well since I added you as a temporary owner.

BenHenning commented:
@seanlip I restored some of the ownership that was left to @DubeySandeep. Let me know if you want any of these permanently changed back.

seanlip commented on .github/CODEOWNERS:
```diff
@@ -570,8 +559,10 @@
 # User Data Takeout.
-/core/domain/takeout_*.py @BenHenning @vojtechjelinek
-/core/domain/wipeout_*.py @BenHenning @vojtechjelinek
+# TODO(#11477): Restore codeowners to @BenHenning after 2021-01-04.
```
@BenHenning Please add me to both of these too.

BenHenning replied:
Done.

seanlip reviewed (approved):
Thanks, @BenHenning! Just one comment.

BenHenning committed:
Update CODEOWNERS

Have Sean replace me instead of me just being removed for wipeout/takeout code owners.

seanlip reviewed (approved):
Thanks, LGTM

BenHenning committed:
Introduce global owners & use oppia/ml

Add global owners using oppia/owners, and replace Prasanna with oppia/ml.

DubeySandeep commented on .github/CODEOWNERS:
```diff
@@ -33,6 +33,7 @@
 # files/folders are created in the following directories, these codeowners would
 # be superseded by the relevant codeowners mentioned elsewhere in this file.
 # (Reference: https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/about-code-owners)
+* @oppia/owners
```
I think this way we won't be able to ensure that all the files are assigned to a specific person.

Assume someone added a new file and forgot to add code owner for that file in that case lint test won't catch it as this line covers all the files and we or owners end up owning a lot of files.

Possible solution:
Do not use lines 36-44 in lint checks for checking whether all the files are owned by someone. wdyt? (I think this is an existing issue and we should fix it separately.)

BenHenning replied:
Ah, actually maybe this isn't an issue if we already have a lint check. That tells me a PR can't get merged if its missing code ownership--that seems to solve the same problem, right? In that case, I think I can remove this.

I don't have strong of an opinion at the moment on whether the other blank ownership lines should be removed. I think that has more to do with our general policy around how to organize codeowners & add them over time. I haven't spent much time thinking about that on the web side yet.

BenHenning committed:
Remove global ownership

Remove global ownership since we already have a linter check to verify this case (and more effectively).
